BoxContentSDK/BoxContentSDK/Requests/BOXFolderRequest.m

```diff
@@ -59,64 +80,140 @@ - (BOXAPIOperation *)createOperation
     // of bandwidth/processing time.
     queryParameters[@"limit"] = @"0";
     
-    BOXAPIJSONOperation *JSONoperation = [self JSONOperationWithURL:URL
-                                                         HTTPMethod:BOXAPIHTTPMethodGET
-                                              queryStringParameters:queryParameters
-                                                     bodyDictionary:nil
-                                                   JSONSuccessBlock:nil
-                                                       failureBlock:nil];
-    if ([self.notMatchingEtags count] > 0) {
-        // Set up the If-None-Match header
-        for (NSString *notMatchingEtag in self.notMatchingEtags) {
-            [JSONoperation.APIRequest addValue:notMatchingEtag
-                            forHTTPHeaderField:BOXAPIHTTPHeaderIfNoneMatch];
+    if (self.associateId != nil) {
+        BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:URL
+                                                             HTTPMethod:BOXAPIHTTPMethodGET
+                                                  queryStringParameters:queryParameters
+                                                         bodyDictionary:nil
+                                                           successBlock:nil
+                                                           failureBlock:nil
+                                                            associateId:self.associateId];
+        
+        NSString *requestDirectory = [BOXContentClient defaultClient].tempCacheDir;
```

Comment:
Also, this makes an incorrect assumption that the default client is always the one used. This kind of indicator confirms to us that this is not the right thing and we should have the target path passed in instead.